src/services.py

```diff
@@ -449,3 +454,181 @@ def ban_user(self, my_user_id, other_user_id):
+        with session_scope() as session:
+            last_message = (
+                session.query(self.Chat)
+                .filter_by(chat_room_id=chat_room_id)
+                .order_by(desc("created_at"))
+                .first()
+            )
+            if last_message == None:
+                return {}
+            return last_message.asdict()
+
+    def add_message(self, chat_room_id, message, author_id):
+        with session_scope() as session:
+            chat = Chat(
+                chat_room_id=str(chat_room_id),
+                message=message,
+                author_id=str(author_id),
+            )
+            session.add(chat)
+            session.flush()
+            session.refresh(chat)
+            chat = chat.asdict()
+
+            result = (
+                session.query(self.ChatRoom, self.Buyer, self.Seller)
+                .filter_by(id=chat.get("chat_room_id"))
+                .outerjoin(self.Buyer, self.Buyer.id == self.ChatRoom.buyer_id)
+                .outerjoin(self.Seller, self.Seller.id == self.ChatRoom.seller_id)
+                .all()[0]
+            )
+            chat_room = result[0].asdict()
+            buyer = result[1].asdict()
+            seller = result[2].asdict()
+
+            (author, dealer) = (
+                (seller, buyer) if seller.get("id") == author_id else (buyer, seller)
+            )
+
+            return {
+                "dealerName": dealer.get("full_name"),
+                "dealerId": dealer.get("id"),
+                "createdAt": datetime.timestamp(chat.get("created_at")),
+                "updatedAt": datetime.timestamp(chat.get("updated_at")),
+                "authorName": author.get("full_name"),
+                "authorId": author.get("id"),
+                "message": chat.get("message"),
+                "chatRoomId": chat_room.get("id"),
+            }
+
+    def get_conversation(self, user_id, chat_room_id):
+        with session_scope() as session:
+            results = (
+                session.query(self.ChatRoom, self.Chat, self.Buyer, self.Seller)
+                .filter(self.ChatRoom.id == chat_room_id)
+                .outerjoin(self.Chat, self.Chat.chat_room_id == self.ChatRoom.id)
+                .outerjoin(self.Buyer, self.Buyer.id == self.ChatRoom.buyer_id)
+                .outerjoin(self.Seller, self.Seller.id == self.ChatRoom.seller_id)
+                .all()
+            )
+
+            data = []
+            for result in results:
+                chat_room_result = result[0].asdict()
+                chat_result = result[1].asdict()
+                buyer = result[2].asdict()
+                seller = result[3].asdict()
+                (author, dealer) = (
+                    (seller, buyer) if seller.get("id") == user_id else (buyer, seller)
+                )
+
+                data.append(
+                    {
+                        "dealerName": dealer.get("full_name"),
```

Comment:
Let's use snake case for the keys of this dict to make it consistent with others